spec/support/table_helpers/column.rb

```diff
@@ -0,0 +1,176 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require_relative 'identifier'
+
+module TableHelpers
+  module Column
+    extend Identifier
+
+    def self.for(header)
+      case header
+      when /\A\s*estimated hours/i
+        raise ArgumentError, 'Please use "work" instead of "estimated hours"'
+      when /derived estimated hours/i
+        raise ArgumentError, 'Please use "derived work" instead of "derived estimated hours"'
+      when /derived remaining hours/i
+        raise ArgumentError, 'Please use "derived remaining work" instead of "derived remaining hours"'
+      when /\A\s*remaining hours/i
+        raise ArgumentError, 'Please use "remaining work" instead of "remaining hours"'
+      when /\A\s*work/i
+        Duration.new(header:, attribute: :estimated_hours)
+      when /derived work/i
+        Duration.new(header:, attribute: :derived_estimated_hours)
+      when /\A\s*remaining work/i
+        Duration.new(header:, attribute: :remaining_hours)
+      when /derived remaining work/i
+        Duration.new(header:, attribute: :derived_remaining_hours)
+      when /subject/
+        Subject.new(header:)
+      when /hierarchy/
+        Hierarchy.new(header:)
+      else
+        assert_work_package_attribute_exists(header)
+        Generic.new(header:)
+      end
+    end
+
+    def self.assert_work_package_attribute_exists(attribute)
+      attribute = to_identifier(attribute).to_s
+      return if WorkPackage.attribute_names.include?(attribute)
+
+      raise ArgumentError, "WorkPackage does not have an attribute named #{attribute.inspect}"
+    end
+
+    class Generic
+      include Identifier
+
+      attr_reader :attribute, :title, :raw_header
+
+      def initialize(header:, attribute: nil)
+        @raw_header = header
+        @title = header.strip
+        @attribute = attribute || to_identifier(title)
+      end
+
+      def format(value)
+        value.to_s
+      end
+
+      def cell_format(value, size)
+        format(value).send(text_align, size)
+      end
+
+      def parse(raw_value)
+        raw_value.strip
+      end
+
+      def text_align
+        :ljust
+      end
+
+      def read_and_update_work_packages_data(work_packages_data)
+        work_packages_data.each do |work_package_data|
+          work_package_data => { attributes:, row: }
```

Comment:
I had to look up what this does. Very cool that it works somewhat like a Hash destructuring.

Reply:
[pattern matching](https://docs.ruby-lang.org/en/3.2/syntax/pattern_matching_rdoc.html). I'm still not sure if I like it or not, so I try using it from times to times to get a better feeling about it.